src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -372,6 +359,31 @@ function InitializeFirebaseFunctionsStubs(functionsDir: string): void {
   };
 }
 
+/*
+  @google-cloud/firestore@2.0.0 made a breaking change which swapped relying on "grpc" to "@grpc/grpc-js"
+  which has a slightly different signature. We need to detect the firestore version to know which version
+  of grpc to pass as a credential.
+ */
+async function getGRPCInsecureCredential(frb: FunctionsRuntimeBundle): Promise<any> {
```

Comment:
So just FYI `@google-cloud/firestore` version `2.2.0` came out today which allows the `ssl: false` flag instead of passing in your own gRPC credential.

So if you detect that library, you can skip knowing about any gRPC at all.

Reply:
https://github.com/googleapis/nodejs-firestore/releases/tag/v2.2.0

Reply:
So yeah, I was thinking about this, but it really doesn't help that much until we drop `firebase-admin@7` because we'll still have to route differently based on version, ya know. Once we drop 7 we'll be able to only support `@google-cloud/firestore>=2.20` then life'll be more buttery.

Reply:
@abeisgoat true just good to know where we are headed.

Even better: the NodeJS Admin SDK now respects `FIRESTORE_EMULATOR_HOST` so we could *entirely avoid* monkey-patching the `initalizeApp()` call and just set up the environment and let the SDK handle it.

That would remove a huge amount of complexity for us.

Reply:
agreed, that would be wonderful, but it'll be awhile before we have enough adoption to actually remove the old code. We'll just sit back and wait.